packages/itinerary-body/src/defaults/duration.tsx

```diff
@@ -1,26 +1,37 @@
-import coreUtils from "@opentripplanner/core-utils";
 import React, { ReactElement } from "react";
 import { FormattedMessage } from "react-intl";
 
+import coreUtils from "@opentripplanner/core-utils";
+import { isTransit } from "@opentripplanner/core-utils/lib/itinerary";
 import { defaultMessages } from "../util";
 
 interface Props {
   seconds: number;
+  mode?: string;
+  showApproximatePrefixNonTransitLegs?: boolean;
 }
 
 /**
  * This is a clickable component that summarizes the leg (travel time, stops
  * passed). On click it will expand and show the list of intermediate stops.
  */
-export default function Duration({ seconds }: Props): ReactElement {
+export default function Duration({
+  seconds,
+  mode,
+  showApproximatePrefixNonTransitLegs
+}: Props): ReactElement {
   return (
     <FormattedMessage
       defaultMessage={
         defaultMessages["otpUi.ItineraryBody.common.durationShort"]
       }
       description="Duration in abbreviated hours (if over one hour) and minutes"
       id="otpUi.ItineraryBody.common.durationShort"
-      values={coreUtils.time.toHoursMinutesSeconds(seconds)}
+      values={{
+        ...coreUtils.time.toHoursMinutesSeconds(seconds),
+        approximatePrefix:
+          showApproximatePrefixNonTransitLegs && !isTransit(mode)
+      }}
```

Comment:
Make `approximatePrefix` a prop, rename to `showApproximatePrefix`, and pass `showApproximatePrefixNonTransitLegs && !isTransit(mode)` from `AccessLegBody` render code, so you don't have to pass `mode` to this component. (Also, you would be passing one prop instead of two.)